test/test_compression.py

```diff
@@ -30,8 +35,8 @@ def write_netcdf(filename,zlib,least_significant_digit,data,dtype='f8',shuffle=F
         #compression=''
         #compression=0
         #compression='gzip' # should fail
-    foo2 = file.createVariable('data2',\
-            dtype,('n'),compression=compression,least_significant_digit=least_significant_digit,\
+    foo2 = file.createVariable('data2',
+            dtype,('n'),compression=compression,least_significant_digit=least_significant_digit,  # type: ignore  # mypy doesn't like compression
```

Comment:
could add this to line 29:
```python
compression: CompressionLevel
if zlib:
    compression='zlib'
else ...
```

Reply:
thinking about it... maybe we should add an `int` Union to the `CompressionLevel` definition?
Keep the Literals for documentation.

On the other hand... how many people will actually not use this hardcoded but in a variable?

Reply:
> thinking about it... maybe we should add an `int` Union to the `CompressionLevel` definition? Keep the Literals for documentation.
> 
> On the other hand... how many people will actually not use this hardcoded but in a variable?

You bring up a good point.

I think this applies broadly to all the `Literal`s. On the side of making the Union, you get less noise but can miss bad literals (though including the `Literal` can still help with code completion). On the side of only allowing a literal is the most assurance that the type is correct, but with more potential noise/onus on the user to do runtime coercion to literals. 

And unfortunately, the type checkers don't support using something like 

```python
assert var in ["abc", "def"]
```

to coerce strings to literals.

I'm willing to make this change now, or wait until the next version if there are enough complaints about the noise in their type checkers.

Reply:
Yeah, my guess is that >95% of people will use inlined expressions for these kwargs or not even use them at all.

So let's wait until someone complains.